File: lab_3_assignment_3/tower_of_hanoi.py
def tower_of_hanoi(num, source, destination, temporary):
    if num == 1:
        print(f"Move Disc {num} from {source} to {destination}")
        return
    tower_of_hanoi(num - 1, source, temporary, destination)
    print(f"Move Disc {num} from {source} to {destination}\n")
    tower_of_hanoi(num - 1, temporary, destination, source)


num = int(input("Enter the number of discs : "))
tower_of_hanoi(num, "A", "C", "B ")

File: lab_3_assignment_3/tower_of_hanoi.c
#include <stdio.h>
int no_counts;

void tower_of_hanoi(int num, char source, char dest, char temp);

int main()
{
    int num;
    char x = 'A', y = 'B', z = 'C';
    printf("Enter a number of discs : ");
    scanf("%d", &num);
    tower_of_hanoi(num, x, z, y);
    printf("Number of moves: %d\n", no_counts);

    return 0;
}

void tower_of_hanoi(int num, char source, char dest, char temp)
{
    no_counts = no_counts + 1;
    if (num == 1)
    {
        printf("Move disc 1 from %c to %c \n", source, dest);
        return;
    }
    tower_of_hanoi(num - 1, source, temp, dest);
    printf("move disc %d from %c to %c \n", num, source, dest);

    tower_of_hanoi(num - 1, temp, dest, source);
}